Libraries/LibCore/File.cpp

```diff
@@ -129,5 +130,16 @@ String File::real_path_for(const String& filename)
     return real_path;
 }
 
+String File::read_link(const String& filename)
+{
+    char buffer[PATH_MAX];
+    int length = readlink(filename.characters(), buffer, sizeof(buffer));
+    if (length <= 0) {
+        return {};
```

Comment:
Is a zero-length `readlink()` result actually a failure? A null `String` result should indicate failure.
Note that `String()` and `String("")` are not the same thing. The former is null, the latter is not.

Reply:
True, ```length``` is not possible to be 0 (can symlink point to zero-length file path?)

Reply:
Yes it can

Reply:
ok